README.md

```diff
@@ -82,7 +84,210 @@ Requirements:
        ```
 
 Build:
-------
+-----------------
+  Aladdin currently supports two ways to build.
+
+  1. Configure with CMake, then build with Make :
+
+       If you are familiar with CMake, or you are interested in a script
+       which can build out-of-source and offer automatical LLVM/Clang
+       installation, choose [CMake](#build-with-cmake)
+
+  2. Directly build with Makefile :
+
+       If you want to keep all the things simple, or you do not have
+       CMake, choose [Makefile](#build-with-makefile)
+
+
+Build with CMake:
+-----------------
+CMake is a configure tool which allows you to out-of-source build.
+Aladdin Requires CMake newer than 2.8.12. By default, CMake
+searches for LLVM 3.4.
+
+
+1. Set `$BOOST_ROOT`,  `TRACER_HOME`, `LLVM_HOME` to let Aladdin know where the tools and libraries are.
+
+   ```
+   export BOOST_ROOT=/your/path/to/boost
+   export TRACER_HOME=/your/path/to/LLVM-Tracer
+   export LLVM_HOME=/your/path/to/llvm
+   ```
+
+2. Use CMake to configure Aladdin and build libraries and executables
+
+   ```
+   mkdir /where/you/want/to/put/your/binary
+   cd /where/you/want/to/put/your/binary
+   cmake /your/path/to/Aladdin
+   make -j4
+   ```
+
+3. (Optional) Run tests with `ctest` command
+
+   **CAUTION : Takes a long time**
+
+   **Suggest running part of them, like `"triad"` or `"unit_test"`.**
+
+   There are 1397 tests. On Intel Xeon E5-2650 2.60GHz,
+   execution in 32 jobs takes 39 minutes,
+   and execution in sequential takes 6.5 hours.
+
+
+   Runs all the tests in sequential.
+   ```
+   ctest
+   ```
+
+   Runs all the tests in parallel. You can mix this option with test
+   filtering option to speed up tests.
+   ```
+   ctest -j JOB_NUM
+   ```
+
+   Runs `triad` tests. Use `-R` flag to denotes the regular expression to match.
+   ```
+   ctest -R triad
+   ```
+
+   Runs unit-tests
+   ```
+   ctest -R unit_test
+   ```
+
+   If just want to know what -R `triad` matches. There is no actual execution of tests
```

Comment:
I added the following line just before `Runs `triad` tests. Use `-R` flag to denotes....`
```
Each test has name and the shell command. the name of test can be used
to filter out the unwanted tests.
```